# Perhitungan BMR

Tujuan dari perhitungan nilai BMR ini adalah untuk menentkan nilai BMR yang menjadi nilai energi kebutuhan dasar untuk pembuatan makan

## Objective

- Menentukan nilai BMR untuk setiap gender
- Menentukan rata-rata berat badan laki-laki dengan umur >=24 tahun dan umur <=35 tahun

### Data Identification and Data Overview

In [1]:
# initiation

import plotly as py
from plotly import graph_objs as go
from plotly.offline import iplot, init_notebook_mode
from plotly import tools

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

init_notebook_mode(connected=True)

In [2]:
# Loading data

data_bmr = pd.read_csv("BMR.csv")

data_bmr

Age  Gender  Height (cm)  Weight (kg)  Index
0     21    Male          162           58      2
1     44  Female          188           80      2
2     45    Male          177           61      2
3     50    Male          192          101      3
4     28  Female          195           61      1
..   ...     ...          ...          ...    ...
145   33    Male          142           71      4
146   32  Female          162           58      2
147   29    Male          174           54      1
148   33  Female          197           72      2
149   23    Male          198           50      0

[150 rows x 5 columns]

In [3]:
data_bmr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Age          150 non-null    int64 
 1   Gender       150 non-null    object
 2   Height (cm)  150 non-null    int64 
 3   Weight (kg)  150 non-null    int64 
 4   Index        150 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 6.0+ KB


In [4]:
data_bmr.head(10)

Age  Gender  Height (cm)  Weight (kg)  Index
0   21    Male          162           58      2
1   44  Female          188           80      2
2   45    Male          177           61      2
3   50    Male          192          101      3
4   28  Female          195           61      1
5   25  Female          180           58      1
6   20  Female          188          115      4
7   25    Male          152          132      5
8   48    Male          174          107      4
9   29    Male          182           70      2

In [5]:
data_bmr.tail(10)

Age  Gender  Height (cm)  Weight (kg)  Index
140   37  Female          177           61      2
141   38    Male          199          156      4
142   35    Male          154          111      5
143   24  Female          172           67      2
144   29  Female          169          103      4
145   33    Male          142           71      4
146   32  Female          162           58      2
147   29    Male          174           54      1
148   33  Female          197           72      2
149   23    Male          198           50      0

In [6]:
data_bmr.shape

(150, 5)

In [7]:
data_bmr.describe().T

count        mean        std    min     25%    50%    75%    max
Age          150.0   34.086667   8.233942   20.0   27.00   34.0   40.0   50.0
Height (cm)  150.0  175.526667  14.988450  141.0  164.25  179.0  188.0  199.0
Weight (kg)  150.0   78.666667  27.421068   50.0   58.00   70.0   89.5  158.0
Index        150.0    2.320000   1.377285    0.0    2.00    2.0    3.0    5.0

In [8]:
data_bmr["Gender"].unique()

array(['Male', 'Female'], dtype=object)

Berdasarkan hasil overview data diperoleh : 
- data BMR terdiri dari 150 baris dan 5 kolom
- tidak terdapat missing value dalam data ini
- kolom "Gender" terdiri dari Male dan Female sebagai object

Keterangan : 
- Age: Usia
- Gender: Jenis Kelamin
- Height (cm): Tinggi Badan
- Weight (kg): Berat badan
- Index: Pengelompokkan berdasarkan BMI

### Data Exploration and Data Processing

Tahapan pertama yang kita akan lakukan adalah dengan menghitung nilai BMI dari masing-masing umur dan gender

In [9]:
data_bmr

Age  Gender  Height (cm)  Weight (kg)  Index
0     21    Male          162           58      2
1     44  Female          188           80      2
2     45    Male          177           61      2
3     50    Male          192          101      3
4     28  Female          195           61      1
..   ...     ...          ...          ...    ...
145   33    Male          142           71      4
146   32  Female          162           58      2
147   29    Male          174           54      1
148   33  Female          197           72      2
149   23    Male          198           50      0

[150 rows x 5 columns]

- nilai_bmr untuk laki-laki = 66 + (13.7 * Weight(kg) ) + (5 * Height(cm) ) - (6.8 * Age)
- nilai_bmr untuk perempuan = 65.5 + (9.6 * Weight(kg) ) + (1.8 * Height(cm) ) - (4.7 * Age)

In [10]:
kolom_bmr = []

for a in range(0,len(data_bmr)):
    if data_bmr.loc[a,"Gender"] == "Male":
        nilai_bmr = 66+(13.7*data_bmr.loc[a,"Weight (kg)"])+(5*data_bmr.loc[a,"Height (cm)"])-(6.8 * data_bmr.loc[a,"Age"])
    else:
        nilai_bmr = 65.5+(9.6 * data_bmr.loc[a,"Weight (kg)"])+(1.8*data_bmr.loc[a,"Height (cm)"])-(4.7*data_bmr.loc[a,"Age"])
    kolom_bmr.append(nilai_bmr)

data_bmr["NIlai BMR"] = kolom_bmr

data_bmr

Age  Gender  Height (cm)  Weight (kg)  Index  NIlai BMR
0     21    Male          162           58      2     1527.8
1     44  Female          188           80      2      965.1
2     45    Male          177           61      2     1480.7
3     50    Male          192          101      3     2069.7
4     28  Female          195           61      1      870.5
..   ...     ...          ...          ...    ...        ...
145   33    Male          142           71      4     1524.3
146   32  Female          162           58      2      763.5
147   29    Male          174           54      1     1478.6
148   33  Female          197           72      2      956.2
149   23    Male          198           50      0     1584.6

[150 rows x 6 columns]

Dari hasil perhitungan diatas bisa dilihat nilai BMR untuk masing-masing data sudah terdapat pada kolom "Nilai BMR"

Selanjutnya adalah kita akan menentukan rata-rata berat badan laki-laki dengan umur >=24 tahun dan umur <=35 tahun

In [11]:
data_bmr.query("Gender == 'Male' & Age >= 24 & Age <=35").mean()

C:\Users\ASUS\AppData\Local\Temp\ipykernel_13532\3767821414.py:1: FutureWarning:

Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.



Age              29.529412
Height (cm)     173.088235
Weight (kg)      86.411765
Index             2.617647
NIlai BMR      1914.482353
dtype: float64

# Conclusion

- NIlai BMR untuk masing-masing data dapat dilihat pada data_bmr terbaru
- Rata-rata untuk berat badan laki-laki dengan umur >= 24 Tahun dan <= 35 Tahun adalah 86.411765 Kg